File: frontend/src/types/task.ts
export type TaskStatus = "PENDING" | "IN_PROGRESS" | "COMPLETED";
export type SubtaskStatus = "PENDING" | "COMPLETED";
export type MetaType = "TEXT" | "LINK" | "COLOR" | "DATE" | "FILE" | "IMAGE";

export interface Jabatan {
  kode: number;
  nama: string;
}

export interface Divisi {
  kode: number;
  nama: string;
}
export interface DivisionSummary {
  kode: number;
  nama: string;
  leaderName: string | null;
  memberCount: number;
  iconKey: string; // <-- ganti avatarUrl
  color: string; // <-- baru
}

export interface Pegawai {
  kode: number;
  nama: string;
  kodejabatan: number;
  kodedivisi: number;
  statusLeader: number;
  jabatan?: Jabatan | null;
  divisi?: Divisi | null;
}

export interface Me {
  kodeku: string;
  username: string;
  avatarUrl?: string | null;
  pegawai?: Pegawai | null;
}

export interface Colleague {
  kodeku: string;
  nama: string;
  statusLeader: number;
  avatarUrl?: string | null;
  jabatan?: Jabatan | null;
}

export interface ReactionSummary {
  emoji: string;
  count: number;
  reacted: boolean;
}

export interface CommentAttachment {
  id: string;
  url: string;
  fileName: string;
  fileType: string | null;
  sizeBytes: number;
}

export interface TaskComment {
  id: string;
  content: string;
  userKode: string;
  createdAt: string;
  parentId: string | null;
  replies: TaskComment[];
  reactions: ReactionSummary[];
  attachments: CommentAttachment[];
}

export interface TaskMeta {
  id: string;
  key: string;
  value: string | null;
  type: MetaType;
  sortOrder: number;
}

export interface Subtask {
  id: string;
  taskId: string;
  description: string;
  status: SubtaskStatus;
  sortOrder: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: TaskStatus;
  userKode: string;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
  sortOrder: number;
  comments: TaskComment[];
  meta: TaskMeta[];
  subtasks?: Subtask[];
}

export interface TaskConnection {
  tasks: Task[];
  nextCursor: string | null;
  hasMore: boolean;
}

// Dipakai saat compose task baru / edit info tambahan, sebelum tersimpan ke server
// ...semua yang lain tetap sama, cuma ini yang berubah:
export interface MetaDraft {
  draftId: string;
  id?: string; // ID asli di database, ada kalau item ini sudah tersimpan (dipakai saat edit)
  key: string;
  value: string;
  type: MetaType;
}

File: backend/internal/models/task.go
// backend/internal/models/task.go
package models

import "time"

type TaskStatus string

const (
	TaskStatusPending    TaskStatus = "pending"
	TaskStatusInProgress TaskStatus = "in_progress"
	TaskStatusCompleted  TaskStatus = "completed"
)

type Task struct {
	ID          uint          `gorm:"primaryKey"`
	Title       string        `gorm:"size:255;not null"`
	Description string        `gorm:"type:text"`
	Status      TaskStatus    `gorm:"size:20;default:pending;not null"`
	UserKode    string        `gorm:"column:user_kode;size:50;not null;index;collate:utf8mb4_general_ci"`
	CreatedBy   string        `gorm:"column:created_by;size:50;not null;index;collate:utf8mb4_general_ci"`
	SortOrder   int           `gorm:"column:sort_order;default:0;index"` // <-- baru, dipakai drag-and-drop
	StartDate   *time.Time    `gorm:"column:start_date"`
	DueDate     *time.Time    `gorm:"column:due_date"`
	CompletedAt *time.Time    `gorm:"column:completed_at"`
	CreatedAt   time.Time     `gorm:"autoCreateTime"`
	UpdatedAt   time.Time     `gorm:"autoUpdateTime"`
	Comments    []TaskComment `gorm:"foreignKey:TaskID"`
	Meta        []TaskMeta    `gorm:"foreignKey:TaskID"`
	Subtasks    []Subtask     `gorm:"foreignKey:TaskID"`
}

func (Task) TableName() string {
	return "xv_task"
}
